tools/KoreBuild.Console/Commands/DockerBuildCommand.cs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace KoreBuild.Console.Commands
+{
+    internal class DockerBuildCommand : SubCommandBase
+    {
+        private const string DockerfileExtension = ".dockerfile";
+        private const string ImagePrefix = "aspnetbuild";
+
+        public CommandArgument Platform { get; set; }
+
+        public List<string> Arguments {get; set; }
+        
+        public string ContainerName { get; set; } = "testcontainer";
+
+        public override void Configure(CommandLineApplication application)
+        {
+            Platform = application.Argument("platform", "The docker platform to run on.");
+            Arguments = application.RemainingArguments;
+
+            base.Configure(application);
+        }
+
+        protected override bool IsValid()
+        {
+            if(string.IsNullOrEmpty(Platform?.Value))
+            {
+                Reporter.Error("Platform is a required argument.");
+                return false;
+            }
+
+            return true;
+        }
+
+        protected override int Execute()
+        {
+            var dockerFileSource = GetDockerFileSource(Platform.Value);
+            var dockerFileDestination = Path.Combine(RepoPath, GetDockerFileName(Platform.Value));
+
+            File.Copy(dockerFileSource, dockerFileDestination, true);
+
+            // If our ToolSource isn't http copy it to the docker context
+            var dockerToolsSource = ToolsSource;
+            string toolsSourceDestination = null;
+            if (!ToolsSource.StartsWith("http"))
+            {
+                dockerToolsSource = "ToolsSource";
+                toolsSourceDestination = Path.Combine(RepoPath, dockerToolsSource);
+                DirectoryCopy(ToolsSource, toolsSourceDestination);
+            }
+
+            try
+            {
+                var buildArgs = new List<string> { "build" };
+
+                var tag = $"{ImagePrefix}:{ContainerName}";
```

Comment:
First off: we shouldn't mix the container name into the image name, it can add confusion.

Docker image tag names are generally in the form: `[owner]/[name]:[variant]`. Here, `owner` should map to an account on Docker Hub (eventually, doesn't have to right now), I think "aspnetbuild" is a good name. `name` generally refers to the content of the image (regardless of base OS, etc.), so `korebuild` or `build` would be good names. `variant` is used to distinguish between versions and OSes usually. Take a look at the `microsoft/dotnet` image for some examples: https://hub.docker.com/r/microsoft/dotnet/

I'd suggest something like `aspnetbuild/korebuild:jessie`. We may even want to get in to version number tags at some point, but I think we can start with this, since we're always building the container "fresh" (using caches) rather than pulling from Docker Hub.

Reply:
Combined with the above comment re:"Platform", this string would become something like `var tag = $"{ImagePrefix}:{Image}` and `ImagePrefix` becomes `aspnetbuild/korebuild`